### DISCUSSION TITLE
LDAP Password required in Laravel but not in non-laravel version

### DISCUSSION THREAD

**Discussion by chrispian** [DirectoryTree/LdapRecord#502] [Q&A] [answered]
I'm having a couple of issues that I'm struggling with, this is the first one. I setup the stand alone version OUTSIDE of laravel and can connect just fine to our LDAP server. Our LDAP/AD server is *not* protected by a password and instead is using network security to protect it. 

In the NON Laravel version when I set the password to blank, it works.

I installed in a Laravel APP and when I do the ldap test command it shows TRUE, that it connected. Then I tried in tinker after installing using this:

```
use LdapRecord\Container;

$connection->auth()->attempt('USER NAME GOES HERE', "");
```

And this returns `A password must be specified.` which is from src/Auth/Guard.php in the attempt() method. 

Is there an easy way to override this behavior in the Laravel version? It would be nice if the PW wasn't required like in the NON Laravel version.

If I comment out the password check in `Auth.php` I am able to connect just fine so I need to be able to do this without a password inside a laravel app. I have the `ldap.php` file setup with the password blank there already.

**Comment by stevebauman** [comment, marked as answer]
Hi @chrispian,

You must use the `bind()` method to anonymously connect to your LDAP server and provide `null` as the password:

```php
use LdapRecord\Container;

$connection->auth()->bind('USER NAME GOES HERE', null);
```

The `attempt()` method is meant to be provided to end-users so that blank passwords cannot be provided to allow anonymous binding.

Let me know if you encounter anything else 👍

**Reply by stevebauman** [reply]
Also, if you're looking to connect anonymously using LdapRecord-Laravel, you must provide `null` as the password in your `.env`, or your `config/ldap.php` file (whichever you're using to configure your connection):

**.env**:

```env
LDAP_USERNAME="cn=john doe,dc=local,dc=com"
LDAP_PASSWORD=null
```

**config/ldap.php**:

```php
'default' => [
    // ...
    'username' => 'cn=john doe,dc=local,dc=com',
    'password' => null,
],
```

**Comment by chrispian** [comment]
@stevebauman Perfect, thank you! That works. Derp, I almost tried that but got distracted and must have thought I tried it. Appreciate the quick reply!

**Reply by stevebauman** [reply]
Happy to help @chrispian ❤️
